packages/experimental/Expander/windows/ReactNativeExpander/ExpanderView.cpp

```diff
@@ -42,210 +42,206 @@ namespace winrt::ReactNativeExpander::implementation {
+                    this->SetResourceColor(L"ExpanderChevronPressedForeground", propertyValue);
+                }
+                else if (propertyName == "chevronBorderThickness") {
                     auto resDict = expander.Resources();
-                    Windows::UI::Xaml::Thickness thickness = Windows::UI::Xaml::ThicknessHelper::FromUniformLength(propertyValue.AsDouble());
-                    resDict.Insert(winrt::box_value(L"ExpanderChevronBorderThickness"), winrt::box_value(thickness));
+                    if (!propertyValue.IsNull()) {
+                        auto thickness = xaml::ThicknessHelper::FromUniformLength(propertyValue.AsDouble());
+                        resDict.Insert(winrt::box_value(L"ExpanderChevronBorderThickness"), winrt::box_value(thickness));
+                    } else {
+                        resDict.Remove(winrt::box_value(L"ExpanderChevronBorderThickness"));
+                    }
+                }
+                else if (propertyName == "chevronBorderBrush") {
+                    this->SetResourceColor(L"ExpanderChevronBorderBrush", propertyValue);
+                }
+                else if (propertyName == "chevronBorderPointerOverBrush") {
+                    this->SetResourceColor(L"ExpanderChevronBorderPointerOverBrush", propertyValue);
+                }
+                else if (propertyName == "chevronBorderPressedBrush") {
+                    this->SetResourceColor(L"ExpanderChevronBorderPressedBrush", propertyValue);
                 }
-            }
-            else if (propertyName == "chevronBorderBrush") {
-                this->SetResourceColor(propertyValue, L"ExpanderChevronBorderBrush");
-            }
-            else if (propertyName == "chevronBorderPointerOverBrush") {
-                this->SetResourceColor(propertyValue, L"ExpanderChevronBorderPointerOverBrush");
-            }
-            else if (propertyName == "chevronBorderPressedBrush") {
-                this->SetResourceColor(propertyValue, L"ExpanderChevronBorderPressedBrush");
             }
         }
         m_updating = false;
     }
 
-    void ExpanderView::SetResourceColor(Microsoft::ReactNative::JSValue const& value, hstring key) {
-        if (!value.IsNull()) {
-            auto expander = (this->Content()).try_as<Microsoft::UI::Xaml::Controls::Expander>();
-            auto color = value.To<winrt::Brush>();
+    void ExpanderView::SetResourceColor(hstring key, Microsoft::ReactNative::JSValue const& value) {
+        if (auto expander = this->Content().try_as<Microsoft::UI::Xaml::Controls::Expander>()) {
             auto resDict = expander.Resources();
-            resDict.Insert(winrt::box_value(key), color);
+            if (!value.IsNull()) {
+                auto color = value.To<winrt::Brush>();
+                resDict.Insert(winrt::box_value(key), color);
+            } else {
+                resDict.Remove(winrt::box_value(key));
```

Comment:
making sure the resource dictionary structure won't throw an exception while retireving an element that's not there

Reply:
this might require "tapping" the app to re-calculate colors/brushes, see https://github.com/microsoft/microsoft-ui-xaml/issues/5457 and https://github.com/microsoft/microsoft-ui-xaml/issues/4443